collections/_utils.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { Predicate } from "./types.ts";
+
+/**
+ * Filters the given array, removing all elements that do not match the given predicate
+ * **in place. This means `array` will be modified!**.
+ */
+export function filterInPlace<T>(
```

Comment:
Since this function is used to implement some of the other functions it would be nice to have some tests for it.

Reply:
As long as it is internal I personally do not agree - testing implementation details instead of the stable contract leads to maintenance effort that does not create value in my experience (see [this talk](https://youtu.be/EZ05e7EMOLM) for a way better delivery of that point).

My opinion here is not strong though, so I am open to add tests if that is what is needed to bring it live^^